base/linalg/symmetric.jl

```diff
@@ -531,50 +531,54 @@ function svdvals!(A::RealHermSymComplexHerm)
 end
 
 # Matrix functions
-function ^(A::Symmetric{T}, p::Integer) where T<:Real
+^(A::Symmetric{<:Real}, p::Integer) = sympow(A, p)
+^(A::Symmetric{<:Complex}, p::Integer) = sympow(A, p)
+function sympow(A::Symmetric, p::Integer)
```

Comment:
This is just to avoid method ambiguities. I tried to `Union`ize this but with no success.